sense-android-library/src/nl/sense_os/datastorageengine/DataSyncer.java

```diff
@@ -162,7 +166,11 @@ protected void downloadSensorsFromRemote() throws IOException, JSONException, Se
         if (sensorList.length() != 0) {
             for (int i = 0; i < sensorList.length(); i++) {
                 JSONObject sensorFromRemote = sensorList.getJSONObject(i);
-                SensorOptions sensorOptions = new SensorOptions(sensorFromRemote.getJSONObject("meta"), false, true, false);
+                // get the default sensor options
+                SensorOptions sensorOptions = DefaultSensorOptions.getSensorOptions(mContext, sensorFromRemote.getString("source_name"));
+                sensorOptions.setMeta(sensorFromRemote.getJSONObject("meta"));
+
+                // if the sensor does not exist create it with the default options
```

Comment:
@tedschmidt i thought we already have getSensor method that will create one if the sensor is not exists, don't we?